src/components/Badge/Badge.tsx

```diff
@@ -13,14 +13,41 @@
  * limitations under the License.
  */
 
-import React from 'react';
-import PropTypes from 'prop-types';
-import styled from '@emotion/styled';
+import React, {
+  forwardRef,
+  MouseEvent,
+  KeyboardEvent,
+  Ref,
+  HTMLProps
+} from 'react';
 import { css } from '@emotion/core';
 import isPropValid from '@emotion/is-prop-valid';
 
+import styled, { StyleProps } from '../../styles/styled';
 import { subHeadingKilo, focusOutline } from '../../styles/style-helpers';
 
+type OnClick = (
+  event: MouseEvent<HTMLDivElement> | KeyboardEvent<HTMLDivElement>
+) => void;
+type RefType = Ref<HTMLDivElement>;
+
+export interface BadgeProps extends HTMLProps<HTMLDivElement> {
+  /**
+   * Callback for the click event.
+   */
+  onClick?: OnClick;
+  /**
+   * Ensures text is centered and the badge looks like a circle.
+   */
+  circle?: boolean;
+  color?: 'neutral' | 'primary' | 'success' | 'warning' | 'danger';
+  /**
+   * The ref to the html div DOM element
+   */
+  ref?: RefType;
+  as?: string;
```

Comment:
This is also missing a description. I suggest matching what we have elsewhere:

```typescript
  /**
   * The HTML element to render.
   */
```

I wonder why we expose `as` in this case, do we have use cases for using a Badge as something else than a div?